bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java

```diff
@@ -221,17 +240,7 @@ private synchronized void updateDailyWeeklyData() {
             cachedDailyData.getValue().ifPresentOrElse(values -> {
                 Aggregate days = values.aggregats.days;
                 updateKwhChannel(YESTERDAY, days.datas.get(days.datas.size() - 1));
-                int idxLast = days.periodes.get(days.periodes.size() - 1).dateDebut.get(weekFields.dayOfWeek()) == 7 ? 2
-                        : 1;
-                Aggregate weeks = values.aggregats.weeks;
-                if (weeks.datas.size() > idxLast) {
-                    updateKwhChannel(LAST_WEEK, weeks.datas.get(idxLast));
-                }
-                if (weeks.datas.size() > (idxLast + 1)) {
-                    updateKwhChannel(THIS_WEEK, weeks.datas.get(idxLast + 1));
-                } else {
-                    updateKwhChannel(THIS_WEEK, 0.0);
-                }
+                setCurrentAndPrevious(values.aggregats.weeks, THIS_WEEK, LAST_WEEK);
```

Comment:
I would like to be sure that the proper week values are delivered every day of the week. I previously implemented it to have a proper 0 value each Monday as current week value and not the value from last week.
So before merging, I will check if the week values are correct Saturday (today), Sunday, Monday and Tuesday.

Reply:
No problem, I'm not in a hurry.

Reply:
Values for yesterday, current week, last week, current month and last month are correct today.
Let's see tomorrow.